PVTPackage/source/MultiphaseSystem/MultiphaseSystemProperties/DeadOilFlashMultiphaseSystemProperties.cpp

```diff
@@ -17,26 +17,55 @@
 #include "pvt/pvt.hpp"
 
 #include <cmath>
+#include <algorithm>
 
 namespace PVTPackage
 {
 
-DeadOilFlashMultiphaseSystemProperties::DeadOilFlashMultiphaseSystemProperties( std::size_t nComponents )
+DeadOilFlashMultiphaseSystemProperties::DeadOilFlashMultiphaseSystemProperties( std::vector< pvt::PHASE_TYPE > const & phases )
   :
-  BlackOilDeadOilMultiphaseSystemProperties( nComponents )
+  BlackOilDeadOilMultiphaseSystemProperties( phases )
 {
+  size_t const nComponents = phases.size();
+  
   for( pvt::PHASE_TYPE pt: getPhases() )
   {
     m_lnFugacity.insert( { pt, pvt::VectorPropertyAndDerivatives< double >( nComponents, nComponents ) } );
   }
 
-  m_moleComposition.at( pvt::PHASE_TYPE::OIL ).value = { 1., 0., 0. };
-  m_moleComposition.at( pvt::PHASE_TYPE::GAS ).value = { 0., 1., 0. };
-  m_moleComposition.at( pvt::PHASE_TYPE::LIQUID_WATER_RICH ).value = { 0., 0., 1. };
+  size_t const nPhases = phases.size();
```

Comment:
So same question: `nComponents` is meant to be equal to `nPhases`, that's it?

Reply:
Yes, for `DeadOil` the number of phases and the number of components are always equal.